### PR TITLE
Use different random seed for each msprime sim

### PR DESCRIPTION
**Pull request by nspope**
As part of porting the ReLERNN architecture to torch, I've been testing against simulations produced by ReLERNN_SIMULATE, and noticed an issue with independence of simulations. **This is only an issue when the seed is set from the command line.**

- The `info.p` file in each of `train`, `test`, `vali` contains a dict with the parameters used for the simulations, including an array of random seeds, one for each simulation. If the seed is set from the command line, then the arrays all contain a single unique random seed (both within and across train/test/validation sets). The reason is this line: https://github.com/kr-colab/ReLERNN/blob/9eb61ff982955a8b727e95fcdd1a8a78e6318230/ReLERNN/simulator.py#L250
- This single unique seed is used in the msprime invocation for every simulation: https://github.com/kr-colab/ReLERNN/blob/9eb61ff982955a8b727e95fcdd1a8a78e6318230/ReLERNN/simulator.py#L85 Note that if self.seed is `None`, then msprime gets its own seed from CPU noise and everything is fine.
- The simulation parameters (mu and rho) differ across simulations, of course: https://github.com/kr-colab/ReLERNN/blob/9eb61ff982955a8b727e95fcdd1a8a78e6318230/ReLERNN/simulator.py#L240-L248

So when seed is set from the command line, the msprime simulations use the same random seed across all training/testing/validation simulations. Because different parameters are used per simulation, the output haplotypes/positions differ across sims. However, reuse of the same random seed could introduce dependence into simulations-- e.g. they are "less random" than iid simulations from the desired coalescent process.

As a rough check, I replaced the `np.repeat(self.seed, numReps)` bit with `np.random.randint(0, some_large_int, size=(numReps,))` and re-ran the simulation. To reduce other sources of randomness aside from msprime, I didn't use an accessibility mask or phasing error, either with the original or modified code. I then trained the model on each of these datasets without permuting haplotypes for each batch (again, to minimize non-msprime randomness).

Here's what the trained models look like on their respective test sets (rec rates normalised to z-scores):
<p float="left">

</p>

I tried a few different runs, with the same result. When the msprime seed is re-used, training is much quicker (train_loss gets to 0.02-ish within 30 epochs or so; as opposed to 0.1-ish in 100 epochs when unique seeds are used). 

This is crude test, but does suggest there is dependence introduced by reusing the same random seed across msprime sims, and that the neural network can pick up on it. In practice this may not actually lead to overfitting, because there's lots of additional noise injected via random masking / haplotype shuffling / phasing error / etc -- and these perturbations don't reuse a seed as far as I can tell. But regardless, the current behavior seems undesirable and is easy to fix.

### CODE REVIEW

**Comment by andrewkern** [comment]
damn. good catch here @nspope and thanks for the PR
